cmd/new/new.go

```diff
@@ -56,10 +55,14 @@ func CreateCommand() *cobra.Command {
 		return utils.Find(formats, toComplete), cobra.ShellCompDirectiveNoSpace | cobra.ShellCompDirectiveNoFileComp
 	})
 
-	home, _ := homedir.Dir()
 	tmp.Flags().StringVar(&localTemplateCache, "templatepath", "", "location of installed templates")
 	viper.BindPFlag("templatepath", tmp.Flags().Lookup("templatepath")) //nolint:errcheck
-	viper.SetDefault("templatepath", filepath.Join(home, ".pdk", "pct"))
+	dir, err := filepath.Abs((filepath.Dir(os.Args[0])))
```

Comment:
This really threw me reading it. And i expected os.getwd(). Reading this post  on stackoverflow i think this library should probably be used mainly due to its claimed cross platform support https://stackoverflow.com/a/18548449

Reply:
So, the reason that `os.Args[0]` was used is that you can derive the path the `pct` binary lives in from there. If I unzipped the `pct` archive to `/usr/local/puppet` and then called in from within a module root:
```sh
cd ~/projects/my-cool-module
./usr/local/puppet/pct new --list
```
... `os.Getwd()` will return `~/projects/my-cool-module`.

There may be a better way to do this, though. I will look in to it a bit further and if not, I'll add some comments to make this clearer.

Reply:
That library you pointed to seems good. Also seeing this solution that uses `os.Executable()` although it seems to be restricted to Go 1.18 +: https://dev.to/natamacm/how-to-get-the-directory-of-the-currently-running-file-47mg

Reply:
I think you misread `1.8` as `1.18`, because `os.Executable()` is there for me and I'm on `1.16`.

I would prefer using built in parts of the language instead of another package for one line. However this is something we don't expect to use long term so maybe this isn't worth so much thought.